server/src/main/java/org/elasticsearch/common/lucene/Lucene.java

```diff
@@ -159,6 +159,24 @@ public static int getNumDocs(SegmentInfos info) {
         return numDocs;
     }
 
+    /**
+     * Returns the average size in bytes of documents in the index referenced by a given {@link SegmentInfos}.
+     */
+    public static long getAverageDocumentSizeInBytes(SegmentInfos info) throws IOException {
+        long totalDocs = 0;
+        long totalBytes = 0;
+        for (SegmentCommitInfo si : info) {
```

Comment:
is there an easy way to ignore tombstones and soft deletes here?

Reply:
why do you want to exclude soft-deletes?  I think it's ok to ignore the tombstones in this heuristic.

Reply:
I take your response to mean there is no easy way, which means I'm good with the heuristic as is. If there is an easy way to do it, I would consider it to protect against poisonous situations where a lot of index/delete patterns reduces doc size by 50%. Not a big deal, just a thought.

Reply:
no there is not except of iterating the entire index I guess. We can for instance only take segments into account with a low number of deleted documents to reduce the impact but I think we are ok here?

Reply:
I'm good thanks. Also, thinking more about this, one can argue that small tombstone docs actually should reduce the average size (as they are what we retain).